### DISCUSSION TITLE
Tried to run the flash_unit tutorial as a stand alone script and recieved an attribute error

### DISCUSSION THREAD

**Discussion by bjl25** [IDAES/idaes-pse#1262] [Q&A]
I installed IDAES a few days ago and went through all the tutorial content on the Jupyter Notebook with no issues. I then wanted to test run a stand-alone script so copied the Flash_unit model into a stand-alone script and ran it from the conda terminal.

It produced multiple attribute errors such as "ERROR: Rule failed for Expression 'fs.flash.split._Vap_flow_mol_ref' with index 0.0: AttributeError: '_GeneralVarLikeExpressionData' object has no attribute '_expr'" 
and 
  "File ~\AppData\Local\anaconda3\envs\idaesBen\Lib\site-packages\idaes\core\base\var_like_expression.py:68 in set_value if self._expr is None or force:
AttributeError: '_GeneralVarLikeExpressionData' object has no attribute '_expr' "

I also tried the recycling tutorial as well and got the same errors.
[Flash_Trial.txt](https://github.com/IDAES/idaes-pse/files/12517376/Flash_Trial.txt)

**Comment by bjl25** [comment]
Update

Installed optional dependencies because I wanted coolprop and randomly reran the flash file and it worked.

**Comment by andrewlee94** [comment]
@bjl25 This was likely due to a conflict between your version of IDAES and Pyomo - Pyomo has recently deprecated the use of `_expr` and IDAES is only making the required adjustments in the v2.2.0 release (which has not formally released yet - it should be out in the next few weeks).

**Reply by LioSergent** [reply]
Got the same problem and solved it by downgrading to Pyomo 6.6.1 (from 6.6.2), using idaes-pse 2.1.0.

Thanks

**Reply by andrewlee94** [reply]
This should have been resolved with the release of IDAES v2.2.0 so it should no longer be necessary to downgrade Pyomo.

Note that IDAES releases are linked to specific Pyomo releases. Due to ongoing development on both sides, we can only guarantee that a given IDAES version will work with its linked Pyomo version.

**Reply by LioSergent** [reply]
Ah yes. I was able to upgrade to ideas-pse 2.2.0 today. Don't know why neither conda nor pip were finding v2.2.0 yesterday. Probably a miss from my end. Thanks
